Missing dependency while setting up webserver with oidc keycloak following the official tutorial

## Environment Details
* Helidon Version:4.1.3
* Helidon SE
* JDK version: 
```
openjdk 22.0.1 2024-04-16
OpenJDK Runtime Environment Temurin-22.0.1+8 (build 22.0.1+8)
OpenJDK 64-Bit Server VM Temurin-22.0.1+8 (build 22.0.1+8, mixed mode, sharing)
```
* OS: Zorin OS 17.2 Core  64bit
* Docker version (if applicable): Docker version 27.3.1

----------

## Problem Description
I encounter some exception following this this [guide](https://helidon.io/docs/v4/se/guides/security-oidc) while trying to setup keycloak with open id connect in Helidon.

## Steps to reproduce
I followed step by step this [guide](https://helidon.io/docs/v4/se/guides/security-oidc) and when I build and run the application I get this exception:
```
Exception in thread "main" java.lang.NoClassDefFoundError: io/helidon/webclient/api/WebClientConfig$Builder
        at io.helidon.security.providers.oidc.common.OidcConfig$Builder.build(OidcConfig.java:998)
        at io.helidon.security.providers.oidc.common.OidcConfig.create(OidcConfig.java:464)
        at io.helidon.security.providers.oidc.OidcFeature$Builder.config(OidcFeature.java:695)
        at java.base/java.util.Optional.ifPresentOrElse(Optional.java:196)
        at io.helidon.security.providers.oidc.OidcFeature$Builder.config(OidcFeature.java:724)
        at io.helidon.security.providers.oidc.OidcFeature.create(OidcFeature.java:218)
        at io.helidon.examples.quickstart.se.Main.routing(Main.java:59)
        at io.helidon.webserver.WebServerConfigSupport$ListenerCustomMethods.routing(WebServerConfigSupport.java:158)
        at io.helidon.webserver.ListenerConfig$BuilderBase.routing(ListenerConfig.java:277)
        at io.helidon.examples.quickstart.se.Main.main(Main.java:42)
Caused by: java.lang.ClassNotFoundException: io.helidon.webclient.api.WebClientConfig$Builder
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        ... 10 more
```
I added the following dependencies:
```yaml
<dependency>
    <groupId>io.helidon.webclient</groupId>
    <artifactId>helidon-webclient-api</artifactId>
</dependency>
<dependency>
    <groupId>io.helidon.webclient</groupId>
    <artifactId>helidon-webclient-http1</artifactId>
</dependency>
```

Then I got the following exception:
```
Exception in thread "main" io.helidon.common.config.ConfigException: Unknown provider configured. Expected providers with types: [security], but only the following providers are supported: [cors, observe], provider interface: io.helidon.webserver.spi.ServerFeatureProvider, configured service: io.helidon.webserver.spi.ServerFeature
	at io.helidon.builder.api.ProvidedUtil.servicesFromObject(ProvidedUtil.java:231)
	at io.helidon.builder.api.ProvidedUtil.discoverServices(ProvidedUtil.java:168)
	at io.helidon.builder.api.Prototype$ConfiguredBuilder.discoverServices(Prototype.java:122)
	at io.helidon.webserver.WebServerConfig$BuilderBase.preBuildPrototype(WebServerConfig.java:443)
	at io.helidon.webserver.WebServerConfig$Builder.buildPrototype(WebServerConfig.java:655)
	at io.helidon.webserver.WebServerConfig$Builder.build(WebServerConfig.java:662)
	at io.helidon.examples.quickstart.se.Main.main(Main.java:43)
```

Finally I added this dependency:
```yaml
<dependency>
    <groupId>io.helidon.webserver</groupId>
    <artifactId>helidon-webserver-security</artifactId>
</dependency>
```

Now everything seems working, but it is not clear to me why I needed to add `helidon-webclient-api` and `helidon-webclient-http1` while working on a web server. Could you please clarify if this is the right approach?
Cheers

--- a/docs/src/main/asciidoc/mp/guides/security-oidc.adoc
+++ b/docs/src/main/asciidoc/mp/guides/security-oidc.adoc
@@ -233,8 +233,12 @@ cd helidon-quickstart-mp
 Update the pom.xml file and add the following Helidon dependency to the `<dependencies>` section.
 
 [source,xml]
-.Add the following dependency to `pom.xml`:
+.Add the following dependencies to `pom.xml`:
 ----
+<dependency>
+    <groupId>io.helidon.microprofile</groupId>
+    <artifactId>helidon-microprofile-security</artifactId>
+</dependency>
 <dependency>
     <groupId>io.helidon.microprofile</groupId>
     <artifactId>helidon-microprofile-oidc</artifactId>

--- a/docs/src/main/asciidoc/se/guides/security-oidc.adoc
+++ b/docs/src/main/asciidoc/se/guides/security-oidc.adoc
@@ -230,11 +230,25 @@ cd helidon-quickstart-se
 Update the pom.xml file and add the following Helidon dependency to the `<dependencies>` section.
 
 [source,xml]
-.Add the following dependency to `pom.xml`:
+.Add the following dependencies to `pom.xml`:
 ----
+<dependency>
+    <groupId>io.helidon.webserver</groupId>
+    <artifactId>helidon-webserver-security</artifactId>
+</dependency>
 <dependency>
     <groupId>io.helidon.security.providers</groupId>
     <artifactId>helidon-security-providers-oidc</artifactId>
+</dependency>
+----
+
+[source,xml]
+.Remove the `test` scope from `helidon-webclient` dependency
+----
+<dependency>
+    <groupId>io.helidon.webclient</groupId>
+    <artifactId>helidon-webclient</artifactId>
+    <scope>test</scope> <!-- remove this line -->
 </dependency>
 ----
 

--- a/docs/src/main/java/io/helidon/docs/mp/guides/SecurityOidcSnippets.java
+++ b/docs/src/main/java/io/helidon/docs/mp/guides/SecurityOidcSnippets.java
@@ -80,7 +80,7 @@ class SecurityOidcSnippets {
                 .path("greet")
                 .request()
                 .header(HttpHeaders.AUTHORIZATION, "Basic " + encoding)
-                .get(GreetingMessage.class);
+                .get(Message.class);
 
         assertThat(jsonMessage.getMessage(), is("Hello World!"));
         // end::snippet_4[]

--- a/docs/src/main/java/io/helidon/docs/mp/guides/SecurityOidcSnippets.java
+++ b/docs/src/main/java/io/helidon/docs/mp/guides/SecurityOidcSnippets.java
@@ -36,12 +36,12 @@ import static org.hamcrest.Matchers.is;
 class SecurityOidcSnippets {
 
     // stub
-    static GreetingMessage createResponse(String str) {
+    static Message createResponse(String str) {
         return null;
     }
 
     // stub
-    record GreetingMessage() {
+    record Message() {
         String getMessage() {
             return "";
         }
@@ -51,7 +51,7 @@ class SecurityOidcSnippets {
     @Authenticated
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public GreetingMessage getDefaultMessage() {
+    public Message getDefaultMessage() {
         return createResponse("World");
     }
     // end::snippet_1[]
@@ -76,7 +76,7 @@ class SecurityOidcSnippets {
     void snippet_6(WebTarget target) {
         // tag::snippet_4[]
         String encoding = Base64.getEncoder().encodeToString("jack:changeit".getBytes());
-        GreetingMessage jsonMessage = target
+        Message jsonMessage = target
                 .path("greet")
                 .request()
                 .header(HttpHeaders.AUTHORIZATION, "Basic " + encoding)

